src/modules/roc_core/optional.h

```diff
@@ -46,6 +46,12 @@ template <class T> class Optional : public NonCopyable<> {
         ptr_ = ptr;
     }
 
+    //! Delete the pointed object and reset the structure
+    void reset() {
```

Comment:
With this addition, the behavior of reset method becomes a bit inconsistent.

This is now legal:

```
foo.reset(some_ptr);
foo.reset();
foo.reset(another_ptr);
```

Bit this is still not:

```
foo.reset(some_ptr);
foo.reset(NULL);
foo.reset(another_ptr);
```

And this is not legal too:

```
foo.reset(some_ptr);
foo.reset(another_ptr);
```

If we're adding support for resetting multiple times, we should adjust the existing reset(T*) version as well and allow all of the examples above.

Actually, after doing it, we don't event need reset() without arguments, we can just use reset(NULL).

Reply:
Cool, thanks for your review

I had to add this **ptr &&** in optional.h line 43
```
if (ptr && (void*)ptr != storage_.mem) {
     roc_panic("optional: attempt to set incorrect object");
``
to deal with the situation when we set NULL as pointer and don't panic